src/zarr/core/attributes.py

```diff
@@ -50,6 +50,7 @@ def put(self, d: dict[str, JSON]) -> None:
            >>> attrs
            {'a': 3, 'c': 4}
         """
+        self._obj.metadata.attributes.clear()
```

Comment:
Is this a separate bug fix to the change in `zarr/core/array.py`? If so, it should get it's own changelog entry, and test.

Reply:
This is to keep `put` (which formerly was using the undesired behavior in `update_attributes`) working as expected rather than a new feature or bugfix